hydromt_sfincs/sfincs.py

```diff
@@ -2886,11 +2899,25 @@ def _parse_datasets_dep(self, datasets_dep, res):
                 try:
                     da_elv = self.data_catalog.get_rasterdataset(
                         dataset.get("elevtn", dataset.get("da")),
-                        geom=self.mask.raster.box,
-                        buffer=10,
+                        # bbox=self.mask.raster.box.to_crs(4326).total_bounds,
+                        # buffer=10,
                         variables=["elevtn"],
                         zoom_level=(res, "meter"),
                     )
+
+                    # TODO properly fix this in hydromt core ?
+                    # check if CRS contains multiple sub CRSs (horizontal and vertical)
+                    if da_elv.raster.crs.is_compound:
+                        da_elv.raster.set_crs(da_elv.raster.crs.sub_crs_list[0])
+
+                    # cropping went wrong for very large datasets with crs != 4326 due to cropping with geom
+                    da_elv = da_elv.raster.clip_bbox(
+                        self.mask.raster.box.to_crs(
+                            da_elv.raster.crs.to_epsg()
+                        ).total_bounds,
+                        buffer=10,
+                    )
```

Comment:
Instead of this fix here wouldn't it be better to change the data itself? This is easily done for geotifs with `rio` a command line tool that comes with rasterio, e.g., `rio edit-info --crs EPSG:3857 example.tif` (this also works for vrt or other gdal raster files)

We can also deal with compound crs in the core, but this needs a bit more thinking. Can you make an issue there?